### ISSUE TITLE
SemanticTokensModifier: Add unused/unreachable

### ISSUE THREAD

**Issue by kjeremy** [microsoft/vscode-languageserver-node#604]
I was thinking today how one might implement "grey out unreachable code" and I think we could do it with semantic tokens. Furthermore it's a pretty common operation spanning many languages. As such I think a modifier for "unused" or "unreachable" should be added to the protocol for consistency across languages.

**Comment by Gama11** [comment]
The LSP already supports this via `DiagnosticTag.Unnecessary`. See also https://github.com/microsoft/vscode/issues/92157.

**Comment by dbaeumer** [comment]
@aeschli what it your take on this. I actually do think we should support unused / unreachable code in semantics as well.

**Comment by aeschli** [comment]
If you want to just mark a single token e.g identifier of a variable or parameter declaration as unused, yes, having a unused modifier is feasible.
If you want to mark a whole range as unreachable, then semantic tokens are not the right thing to use. We don't allow overlapping ranges.
From your description it sounds that it's range of unused code...

**Comment by kjeremy** [comment]
Yes I was originally thinking a whole range which is why I proposed "unreachable" but agree that if we are to mark single tokens (and I think that's entirely reasonable and desired) as unused then there should be an "unused" modifier.

**Comment by dbaeumer** [comment]
@aeschli any take of adding the modifier?

**Comment by aeschli** [comment]
I created microsoft/vscode#97063 to collect proposals for new standard token types/modifiers.

**Comment by dbaeumer** [comment]
Thanks @aeschli

**Comment by dbaeumer** [comment]
Lets wait for the outcome in VS Code.

**Comment by starball5** [comment]
Related on Stack Overflow: [How can I disable problem underlines for unused code with the Rust Analyzer VS Code extension and instead highlight it gray?](https://stackoverflow.com/q/76575226/11107541)
